### PR TITLE
176 read rcp dat files

### LINKED ISSUE #176: Support reading concentration files
**Is your feature request related to a problem? Please describe.**

Shouldn't pymagicc be able to read concentration files like `RCP3PD_MIDYEAR_CONCENTRATIONS.DAT` from http://www.pik-potsdam.de/~mmalte/rcps/ ?

```
pymagicc.read_scen_file("data/RCP3PD_MIDYEAR_CONCENTRATIONS.DAT")
```
gives
```
NoReaderWriterError                       Traceback (most recent call last)
<ipython-input-11-aa122a6d6843> in <module>()
----> 1 pymagicc.read_scen_file("data/RCP3PD_MIDYEAR_CONCENTRATIONS.DAT")

~/projects/pymagicc/pymagicc/__init__.py in read_scen_file(filepath)
     52     """
     53     mdata = MAGICCData()
---> 54     mdata.read(filepath)
     55 
     56     return mdata

~/projects/pymagicc/pymagicc/io.py in read(self, filepath)
   1698         """
   1699         _check_file_exists(filepath)
-> 1700         self.metadata, self.df = self._read_and_return_metadata_df(filepath)
   1701 
   1702     def _read_and_return_metadata_df(self, filepath):

~/projects/pymagicc/pymagicc/io.py in _read_and_return_metadata_df(self, filepath)
   1701 
   1702     def _read_and_return_metadata_df(self, filepath):
-> 1703         reader = self.determine_tool(filepath, "reader")(filepath)
   1704         return reader.read()
   1705 

~/projects/pymagicc/pymagicc/io.py in determine_tool(self, filepath, tool_to_get)
   1871             )
   1872 
-> 1873         raise NoReaderWriterError(error_msg)
   1874 
   1875 

NoReaderWriterError: Couldn't find appropriate reader for RCP3PD_MIDYEAR_CONCENTRATIONS.DAT.
The file must be one of the following types and the filepath must match its corresponding regular expression:
SCEN: ^.*\.SCEN$
SCEN7: ^.*\.SCEN7$
prn: ^.*\.prn$
EmisIn: ^.*\_EMIS.*\.IN$
ConcIn: ^.*\_CONC.*\.IN$
OpticalThicknessIn: ^.*\_OT\.IN$
RadiativeForcingIn: ^.*\_RF\.(IN|MON)$
Out: ^DAT\_.*\.OUT$
TempOceanLayersOut: ^TEMP\_OCEANLAYERS.*\.OUT$
BinOut: ^DAT\_.*\.BINOUT$
```
Trying the pymagicc.io code directly with

```
mdata = pymagicc.io.MAGICCData()
mdata.read("data/RCP3PD_MIDYEAR_CONCENTRATIONS.DAT")
```
gives the same error.

**Describe the solution you'd like**

Add support in `pymagicc.io` and provide an example for reading concentration files.

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

Adds the ability to read the data files released as part of the RCP data generation process, hence closes #176

### CODE REVIEW

File: pymagicc/io.py [hunk 1]
```diff
@@ -686,6 +685,91 @@ def _read_notes(self):
         return notes
 
 
+class _RCPDatReader(_InputReader):
+    # not always reading in Emissions but fine for now
```
[the remark this answers is not in the document]
Comment (znicholls) [reply pymagicc/io.py]: Damn it I thought I'd deleted that
```suggestion
```

File: pymagicc/io.py [hunk 2]
```diff
@@ -686,6 +685,90 @@ def _read_notes(self):
...
+
+            column_headers = converter._read_units(column_headers)
+
+        df = self._convert_data_block_and_headers_to_df(stream, column_headers)
+
+        return df, metadata
+
+    def _convert_variables_to_openscm_variables(self, rcp_variables):
+        magicc7_vars = convert_magicc6_to_magicc7_variables(rcp_variables)
+        # only reliable way to check data I think...
```
Comment (rgieseke) [inline pymagicc/io.py]: Care to elaborate?

File: pymagicc/io.py [hunk 3]
```diff
@@ -1974,3 +2058,38 @@ def pull_cfg_from_parameters_out_file(
...
+
+def get_generic_rcp_name(inname):
+    """Convert an RCP name into the generic Pymagicc RCP name
+
+    The conversion is case insensitive.
+
+    Parameters
+    ----------
+    inname : str
+        The name for which to get the generic pymagicc rcp name
```
Comment (rgieseke) [inline pymagicc/io.py]:
```suggestion
         The name for which to get the generic Pymagicc RCP name
```

File: pymagicc/io.py [hunk 4]
```diff
@@ -1974,3 +2058,38 @@ def pull_cfg_from_parameters_out_file(
...
+
+    Returns
+    -------
+    str
+        The generic pymagicc rcp name
```
Comment (rgieseke) [inline pymagicc/io.py]:
```suggestion
        The generic Pymagicc RCP name
```

**Comment by rgieseke** [comment]
Awesome job! Probably good if @matthiasmengel can give this a look too ...

**Comment by znicholls** [comment]
> Probably good if @matthiasmengel can give this a look too

Would be ideal but I think he's away for 6 weeks so let's merge and he can check when back. Once @lewisjared is happy can one of you two hit the button please?

File: pymagicc/definitions/__init__.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 """
 This module contains all of the relevant definitions for handling MAGICC data.
 
-When we store the data in csv's, we use `Data Packages <https://frictionlessdata.io/
+When we store the data in csv's, we use `Data Packagses <https://frictionlessdata.io/
```
Comment (lewisjared) [inline pymagicc/definitions/__init__.py]: Typo

**Pull request merged**